test/dnnl/TestJsonParser.cpp

```diff
@@ -0,0 +1,26 @@
+#include <gtest/gtest.h>
+
+#include "DnnlTestUtils.h"
+
+#include "graph/backend/elyzor/include/dnnl_graph_compiler.h"
+
+static void compile(const char *fileName) {
+  auto json = read_str_resource(fileName);
+
+  const struct dnnl_graph_compiler *gc;
+  const struct dnnl_graph_compiler_executable *exe;
+
+  ASSERT_EQ(dnnl_graph_compiler_create(nullptr, &gc), dnnl_success);
+  ASSERT_EQ(dnnl_graph_compiler_compile(gc, json.c_str(), &exe), dnnl_success);
+
+  dnnl_graph_compiler_destroy_executable(gc, exe);
+  dnnl_graph_compiler_destroy(gc);
+}
+
+TEST(TestJsonParser, AddRelu) { compile("add_relu.json"); }
+TEST(TestJsonParser, Mpl) { compile("mpl.json"); }
```

Comment:
The API does not assume that the module is returned. I'll add the validation when the compilation is implemented.

Reply:
maybe it's worth adding a few simple sanity tests for `JsonParser` class for now?

Reply:
I used prints for testing - https://github.com/intel/graph-compiler/pull/82/commits/6d49850b70a4ac632cd3c7a8750d86fb0a84ad6e#diff-ac4b138cd27d5ef48530efc874e20d37bb483d9b31d16153432f22f48362cea0R68

Any suggestions what could we test here for now? The object returned by the compile() function is empty, because the compilation is not yet implemented.

Reply:
> The object returned by the compile() function is empty, because the compilation is not yet implemented.

I mean to test the `JsonParser` itself and verify that the output mlir module is correct, something as simple as:
```cpp
TEST(TestJsonParser, AllOpsParsed) {
    auto module = JsonParser::parse("graph_with_3_ops.json");

    // Count the number of operations in the module
    int numOps = 0;
    module.walk([&](mlir::Operation *op) {
        ++numOps;
    });
    ASSERT_EQ(numOps, 3);
}

// more ideas:
// - read a json with one op and check that all attributes are preserved
// - verify that the number of input/output for an op matches
// ...
```

Reply:
Done.